lib/internal/cluster/primary.js

```diff
@@ -58,6 +62,19 @@ else if (process.platform === 'win32') {
 
 cluster.schedulingPolicy = schedulingPolicy;
 
+function validateAndReturnScheduler(scheduler, schedulingPolicy) {
+  if (scheduler !== undefined) {
+    if (typeof scheduler.execute !== 'function') {
+      throw new ERR_CLUSTER_INVALID_SCHEDULER('scheduler.execute');
```

Comment:
Not sure this warrants a new error code. `ERR_INVALID_ARG_TYPE` would probably still work here.

Reply:
`INVALID_ARG_TYPE` seems very broad and that's why i made the choice of `ERR_CLUSTER_INVALID_SCHEDULER`. Will update the use to `INVALID_ARG_TYPE`.